src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -4412,6 +4413,16 @@ int RGWRados::fetch_remote_obj(RGWObjectCtx& obj_ctx,
     encode(trace, bl);
     cb.get_attrs()[RGW_ATTR_OBJ_REPLICATION_TRACE] = std::move(bl);
   }
+  {
+    // add x-amz-replicated-at
+    bufferlist bl;
+    ceph::real_time timestamp = real_clock::now();
+    encode(timestamp, bl);
+
+    std::string replicated_time = ceph::to_iso_8601_no_separators(timestamp,
+                                                                  ceph::iso_8601_format::YMDhms);
+    cb.get_attrs()[RGW_ATTR_OBJ_REPLICATION_TIMESTAMP].append(replicated_time);
```

Comment:
the encode part looks good. we should just stuff that bufferlist bl into the xattr, and not build the string at all

Reply:
I don' think the output prints out correctly in that case:
 
`"user.rgw.replicated-at": "��e�bT\u0012",
`

Reply:
separately, the replication trace also doesn't look right:

`"user.rgw.replication-trace": "\u0001",`

we want the source zone id or name here right?

Reply:
is that `object stat` output you're looking at? we're encoding the timestamp as binary, so it won't render nicely as a string there. but you can customize how it's displayed at https://github.com/ceph/ceph/blob/f4758e5/src/rgw/rgw_admin.cc#L8308-L8312

what's most important is that `dump_time()` is formatting the response header correctly